clean_C2 branch Make func and clean version 확인 요망

1000peach opened the pull request:
Make func and clean version  변경 사항

- main에서 게임의 승패여부 나타내는 부분 함수화, 출력 부분 분리 (isWin, isWinPrint)
- 게임 첫 시작 또는 x키로 재시작 시 빈 gameboard에 랜덤 수 2개 등장하는 부분 함수화 (tilesEmptyBoard)
- 함수화 시 구현 확인을 위해 고쳐야 하는 간단한 오류 개선
ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ pull request 이후 수정 사항
- 간단한 승패여부 판단 위해 int win_result 변수 추가
- tilesEmptyBoard 반환형 수정
- clearWindow() 통해 window와 linux에서 전체 화면 지우는 부분 함수화

총 3 commits의 clean_C2 branch 입니다.
전체적으로 반복되는 부분들을 함수화 하였습니다.
pull request에서 요구받은 수정 사항들 개선하여 다시 올립니다.

더 자세한 변경 사항은 commit message에 적어두었습니다.
확인하시고 수정사항 또는 오류가 발생하면 코멘트로 남겨주세요!
@LeeEunhyung @wlals7217

wlals7217 commented:
확인했습니다. 메인함수가 한 층 더 짧아졌네요 훨씬 보기 좋은 것 같습니다. 머지 찬성합니다!

LeeEunhyung commented:
코드 확인했습니다!
isWin 함수 리턴값을 변수에 넣으면 깔끔할 것같아요! 저도 isGameOver 함수 리턴값 변수에 넣는 방식으로 수정하겠습니다!
improve_L2에서 삭제한 clearWindow 함수를 다시 넣는게 좋을 것같아요. 결과창이 계속 갱신되는 방식이라서 해당 함수가 필요할 것같습니다!! 번복해서 죄송해요...
수고하셨습니다!!

1000peach commented:
알겠습니다!
수정사항들 다시 수정해서 pull request 올리겠습니다~

1000peach commented:
> 확인했습니다. 메인함수가 한 층 더 짧아졌네요 훨씬 보기 좋은 것 같습니다. 머지 찬성합니다!

@wlals7217 수정사항 개선하여 다시 pull request 했습니다. 확인 부탁드려요~

1000peach commented:
> 코드 확인했습니다!
> isWin 함수 리턴값을 변수에 넣으면 깔끔할 것같아요! 저도 isGameOver 함수 리턴값 변수에 넣는 방식으로 수정하겠습니다!
> improve_L2에서 삭제한 clearWindow 함수를 다시 넣는게 좋을 것같아요. 결과창이 계속 갱신되는 방식이라서 해당 함수가 필요할 것같습니다!! 번복해서 죄송해요...
> 수고하셨습니다!!

@LeeEunhyung 수정사항 개선하여 다시 pull request 했습니다. 확인 부탁드려요~

wlals7217 commented:
커밋이 11시간전으로 표시되는데 어떻게 된건가요? 세 번째 커밋 지금하신 거 맞지 않나요?

1000peach commented:
> 커밋이 11시간전으로 표시되는데 어떻게 된건가요? 세 번째 커밋 지금하신 거 맞지않나요?

네 맞아요! 항상 작업 전에 가상 머신으로 현재 시간 변경해서 사용하는데, 모종의 이유로 인터넷 브라우저에서는 맞지 않네요 ㅠㅠ
git log 기록은 제대로 시간이 표시되었으니 참고해주시면 될 것 같아요!

LeeEunhyung commented:
어 맞네요! 코드에는 이상 없는 것같아요!
log 기록에서 확인해보겠습니다.
merge 찬성할게요!!

wlals7217 commented:
가상머신 쓰시는군요. 머지찬성합니다!

1000peach commented:
merge 하겠습니다!!

1000peach commented:
v0.4.3 Make func and clean version

1000peach committed:
Make function (isWin, isWinPrint)

- isWin: main에서 게임 승패 여부 판단 부분을 함수화
- isWinPrint: 승패 여부에 따른 상태를 출력
- 함수화 도중 올바른 승패 여부를 검사하기 위해 간단한 오류 수정

1000peach committed:
Make function (tilesEmptyBoard)

- 빈 gameboard에 랜덤 수 2개 생성하는 부분을 함수화

1000peach committed:
Modify requirements from previous commit

- 더 간단한 승패여부 판단 위해 int win_result 변수 추가
- tilesEmptyBoard 반환형 수정
- window, linux에서 화면 지울 때 다른 함수 쓰는 것 수정: clearWindow()
